aws-lc-rs/src/cipher.rs

```diff
@@ -208,15 +161,27 @@ pub enum OperatingMode {
 
 /// The contextual data used to encrypted/decrypt data.
 #[non_exhaustive]
-pub enum DecryptionContext {
+pub enum CipherContext {
     /// A 128-bit Initalization Vector.
-    Iv128(iv::FixedLength<16>),
+    Iv128(FixedLength<16>),
 
     /// No input to the cipher mode.
     None,
 }
 
+const EMPTY_BYTE_ARRAY: [u8; 0] = [];
+
+impl AsRef<[u8]> for CipherContext {
```

Comment:
My only concern with this is if there is a mode whose context can't be defined from a single `&[u8]`. For example are there modes where it would make sense to model this as a tuple of input values etc.

If we want to support such a potential use case in the future then implementing `AsRef<[u8]>` might be problematic.

Reply:
Suggestion:
```rust
impl TryFrom<CipherContext> for &[u8] {
    type Error;

    fn try_from(value: CipherContext) -> Result<Self, Self::Error> {
        todo!()
    }
}
```